skrub/_datetime_encoder.py

```diff
@@ -181,23 +182,27 @@ def fit(self, X: ArrayLike, y=None) -> "DatetimeEncoder":
             Fitted DatetimeEncoder instance (self).
         """
         self._validate_keywords()
-        if isinstance(X, pd.DataFrame):
-            self.col_names_ = X.columns.to_list()
-        else:
-            self.col_names_ = None
-        X = check_input(X)
+        if not hasattr(X, "__dataframe_consortium_standard__"):
+            X = check_input(X)
+            X = pd.DataFrame(X, columns=[str(i) for i in range(X.shape[1])])
+        X = X.__dataframe_consortium_standard__()
+        if hasattr(X, 'collect'):
+            X = X.collect()
```

Comment:
Now this is where it gets properly controversial, because `.collect` isn't part of the Dataframe API

There's a discussion here if you're interested https://github.com/data-apis/dataframe-api/pull/294#discussion_r1375262323 , hope we can find a resolution soon-ish. Any input there would be greatly appreciated

The reason `collect` is necessary here is that, a few lines down, you're calling `float` on a scalar